yarn-project/sequencer-client/src/block_builder/solo_block_builder.ts

```diff
@@ -416,9 +418,14 @@ export class SoloBlockBuilder implements BlockBuilder {
     );
   }
 
-  protected getKernelDataFor(tx: ProcessedTx) {
-    return new PublicKernelData(
-      tx.data,
+  protected getKernelDataFor(tx: ProcessedTx): RollupKernelData {
+    const inputs = new RollupKernelCircuitPublicInputs(
+      tx.data.aggregationObject,
+      CombinedAccumulatedData.recombine(tx.data.endNonRevertibleData, tx.data.end),
```

Comment:
I think this recombination would generate invalid base rollup inputs since it could invalidate the deduplication, since the deduplication is done independently, unless I missed something

Reply:
writes in `endNonRevertibleData` and `end` are squashed independently prior to this as part of `AbstractPhaseManager.removeRedundantPublicDataWrites` which is performed at the end of each phase, but then in `recombine` here, we squash across phases. So I think it is necessary to produce a valid deduplication.

Reply:
Ah true, I missed the reduce that was doing the deduplication... But It's not doing r/w squashing right? so if in revertible I do a read that needs to be squashed against a nonRevertible write, is it going to squash those two together in recombine?

Reply:
Ahhh but I think it's fine since we are still not validating reads (since reads should be validated in the public kernel eventually IIRC)

Reply:
Cool cool!